### PR TITLE
adds metrics data models and some tests

### PR DESCRIPTION
**Pull request by lspangler584**
Adds Metrics classes and tests, including serialization and transport.

### CODE REVIEW

File: src/NewRelic.Telemetry/Metrics/Metric.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        [DataMember(Name = "value")]
+        public abstract object MetricValue { get; }
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        public Dictionary<string, object> Attributes { get; set; }
+
+        protected Metric()
```
[the remark this answers is not in the document]
Comment (lspangler584) [reply src/NewRelic.Telemetry/Metrics/Metric.cs]: Gone

File: src/NewRelic.Telemetry/Metrics/Metric.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        public double? Min { get; set; }
+        public double? Max { get; set; }
+    }
+
+    public abstract class Metric
+    {
+        /// <summary>
+        /// TODO
+        /// </summary>
+        public string Name { get; set; }
```
[the remark this answers is not in the document]
Comment (lspangler584) [reply src/NewRelic.Telemetry/Metrics/Metric.cs]: no
fixd

**Comment by lspangler584** [comment]
Merging even tho flakey test failed. Fixed in https://github.com/newrelic/newrelic-telemetry-sdk-dotnet/pull/52.

File: src/NewRelic.Telemetry/Metrics/Metric.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+{
+    [DataContract]
+    public class MetricSummaryValue
+    {
+        //TODO:  Figure out how a caller instantiates one of these
+
+        public double Count { get; set; }
+        public double Sum { get; set; }
+        public double? Min { get; set; }
+        public double? Max { get; set; }
```
Comment (jifeingo) [inline src/NewRelic.Telemetry/Metrics/Metric.cs]: Why are count and sum not nullable, but min and max are?

**Pull request merged**